### PR TITLE
fix: address post-merge review findings

### PR DESCRIPTION
**Pull request by albert-einshutoin**
## Summary
- Fix copyable policy recipes so API/download prefixes match runtime path-prefix semantics, and avoid Cognito access-token `aud` validation in the access-token recipe
- Align disabled `firewall.challenge` handling in capabilities/readiness with compiler behavior
- Harden AWS origin-request JWKS DNS checks and HMAC body-hash signing fail-closed behavior
- Update rest-api archetype and golden fixtures for the corrected `/api` prefix

## Verification
- `npm run build:ts`
- `node scripts/programmatic-api-unit-tests.js`
- `node scripts/compile-unit-tests.js`
- `EDGE_ADMIN_TOKEN=ci-build-token-not-for-deploy node scripts/runtime-tests.js`
- `EDGE_ADMIN_TOKEN=ci-build-token-not-for-deploy npm run test:unit`
- `EDGE_ADMIN_TOKEN=ci-build-token-not-for-deploy npm run test:runtime`
- `EDGE_ADMIN_TOKEN=ci-build-token-not-for-deploy npm run lint:policy:ci`
- `EDGE_ADMIN_TOKEN=ci-build-token-not-for-deploy npm run test:drift`
- `EDGE_ADMIN_TOKEN=ci-build-token-not-for-deploy npm run test:all`
- `git diff --check`

### CODE REVIEW

File: templates/aws/origin-request.js [hunk 1]
```diff
@@ -595,6 +603,10 @@ function originAuthBodyHash(request, includeBodyHash) {
   if (!includeBodyHash) return { ok: true, hash: '' };
   const body = request && request.body;
   if (!body || body.data == null) {
+    const method = String((request && request.method) || '').toUpperCase();
+    if (method && !['GET', 'HEAD', 'OPTIONS'].includes(method)) {
+      return { ok: false, error: 'origin_auth_body_unavailable' };
```
Comment (albert-einshutoin) [inline templates/aws/origin-request.js:623]: `include_body_hash: true` でも、`request.body` が無いことは「CloudFront が payload を落とした」ことと同義ではありません。空 payload の `POST`/`PUT`/`PATCH`/`DELETE` は普通にあり、この分岐だとそうした正当なリクエストまで 503 になります。fail-closed にしたいのは body が期待されるのに取得できないケースだけなので、HTTP method ではなく `Content-Length` / `Transfer-Encoding` / body metadata の有無で判定するか、少なくとも empty-body POST regression test を追加して現在の挙動を固定した方がよいです。

File: templates/aws/origin-request.js [hunk 2]
```diff
@@ -595,6 +603,10 @@ function originAuthBodyHash(request, includeBodyHash) {
   if (!includeBodyHash) return { ok: true, hash: '' };
   const body = request && request.body;
   if (!body || body.data == null) {
+    const method = String((request && request.method) || '').toUpperCase();
+    if (method && !['GET', 'HEAD', 'OPTIONS'].includes(method)) {
```
Comment (albert-einshutoin) [inline templates/aws/origin-request.js]: **<sub><sub>P1 Badge</sub></sub>  Do not fail closed on legitimately empty non-GET requests**

This method-only check now rejects every POST/PUT/PATCH/DELETE request when CloudFront omits `request.body`, even if the request is intentionally empty (`Content-Length: 0`). That is stricter than the docs added in this PR, which say to fail only for payload-bearing requests whose body was not included, and it will turn empty webhook handshakes / POST health probes into 503s as soon as `include_body_hash` is enabled. Please distinguish "body unavailable" from "empty body" using the request headers (for example `content-length` / `transfer-encoding`) and add a runtime test for an empty-body POST that should still sign the canonical empty hash.

File: templates/aws/origin-request.js (hunk 1 above)
Comment (albert-einshutoin) [inline templates/aws/origin-request.js:623]: `include_body_hash: true` の fail-closed 方向自体は良いですが、この分岐だと CloudFront が body を付けてこない **正当な bodyless POST/DELETE** まで 503 になります。Lambda@Edge では body 未送信と空 body を区別できないので、ここで method だけを根拠に reject すると、例えば payload なしの POST webhook handshake や DELETE が恒常的に壊れます。空 body を SHA-256("") として扱う従来挙動は残しつつ、`inputTruncated` や `body.action` など CloudFront が実際に body を省略したと判定できる条件だけ fail closed に絞った方がよいです。

File: templates/aws/origin-request.js (hunk 2 above)
Comment (albert-einshutoin) [inline templates/aws/origin-request.js]: This now fails closed for every non-`GET`/`HEAD`/`OPTIONS` request whose `request.body` is absent, but CloudFront can legitimately omit `request.body` when the request is bodyless. With `include_body_hash: true`, an empty `DELETE` or `POST` with `Content-Length: 0` will start returning 503 even though the correct body digest is the empty SHA-256. Please key the failure path off actual payload presence (for example `Content-Length`/`Transfer-Encoding`) rather than HTTP method alone, and add a regression test for an empty-body non-GET request.

Comment (albert-einshutoin) [inline templates/aws/origin-request.js]: This now rejects every non-GET/HEAD/OPTIONS request when CloudFront did not include a body, even if the request legitimately has no payload at all. A `POST`/`PUT` endpoint with query-only semantics or `Content-Length: 0` will start returning `origin_auth_body_unavailable` even though signing the empty body is still correct and was previously supported. The fail-closed check should key off evidence that a payload was actually omitted (for example `body.inputTruncated`, `transfer-encoding`, or a positive `content-length`) rather than the HTTP method alone.

**Review by albert-einshutoin — COMMENTED** [review]
The hardening changes look directionally correct, but one of the new fail-closed branches regresses legitimate empty-body requests.
